### PR TITLE
STYLE: run pre-commit filters on the repo

### PR DESCRIPTION
**Pull request by basnijholt**
I noticed that `pre-commit` is used and ran `pre-commit run --all-files` which results in massive diffs, mostly because of `isort`.

I have added the correct `noqa` and `isort:skip` flags to the lines which needed it to function correctly with the current `.pre-commit-config.yaml`, such that the imports are correctly sorted now.

I also added `seed-isort-config` such that `setup.cfg` is parsed for `pre-commit` and fixed the formatting of `.pre-commit-config.yaml`.

~~Finally, there is `pandas/__init__.py` that is touched by `pre-commit run --all-files`, however, this has less trivial changes and might result in something breaking, so I don't dare to touch it.~~
It was because the `skip` entry in `[isort]` in `setup.cfg` is somehow ignored when running pre-commit and I can't figure out why (I will open an issue in the `isort` repo when it's clear that this will be merged). I have also deleted the other ignored files because I fixed the problems there. `pandas/core/api.py` needs to stay in `skip` because `isort` and `black` make conflicting edits.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
> seed-isort-config

Can you say a bit more about why this is needed / what it does? I'm not familiar with it.

**Comment by jreback** [comment]
are you sure that you have merged the latest master. these should not be necessary.

**Comment by basnijholt** [comment]
@jreback, yes I have just rebased on the latest master.

**Comment by basnijholt** [comment]
@TomAugspurger, `seed-isort-config` statically populates the [`known_third_party`](https://github.com/pandas-dev/pandas/pull/27915/files#diff-380c6a8ebbbce17d55d50ef17d3cf906R119) field, such that `isort` works correctly. Previously `hypothesis` wasn't correctly sorted, even though it was already listed. `seed-isort-config` ensures that all third party libs are there, now `isort` works as expected.

**Comment by TomAugspurger** [comment]
Thanks for the explanation.

Have you checked why the CI is failing? Seems there's an import issue.

**Comment by basnijholt** [comment]
@TomAugspurger there seems to be some kind of circular import. I will fix it! 👍

**Comment by basnijholt** [comment]
I think I fixed it.

For some reason `DataFrame` needs to be imported after `NamedAgg` to avoid a circular import.

**Comment by TomAugspurger** [comment]
Thanks. Looks like a black issue on our CI: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=16280.

**Comment by basnijholt** [comment]
All passes now 👍

**Review by TomAugspurger — APPROVED** [review]
LGTM, thanks.

I'll merge later today to give others another chance to review.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+    hooks:
+    -   id: flake8
+        language: python_venv
+        additional_dependencies: [flake8-comprehensions]
+-   repo: https://github.com/pre-commit/mirrors-isort
+    rev: v4.3.20
+    hooks:
+    -   id: isort
+        language: python_venv
+-   repo: https://github.com/asottile/seed-isort-config
```
[the remark this answers is not in the document]
Comment (sbalk) [reply .pre-commit-config.yaml]: https://github.com/pandas-dev/pandas/pull/27915#issuecomment-521644867

**Comment by TomAugspurger** [comment]
Thanks @sbalk!

**Pull request merged**